### PR TITLE
typings(Attachment): make `attachment` private

### PR DESCRIPTION
**Pull request by net-tech**
**Please describe the changes this PR makes and why it should be merged:**

Removes `<Attachment>.attachment`. This isn't documented anywhere in the [docs](https://discord.js.org/#/docs/discord.js/main/class/Attachment) and from my testing returns a URL like `https://cdn.discordapp.com/ephemeral-attachments/XXXXXXXXXXXXXXXXXX/XXXXXXXXXXXXXXXXXX/message.txt`. If this is here on purpose then the typing should be `string`, not `BufferResolvable | Stream`.

**Status and versioning classification:**

- This PR **only** includes non-code changes, like changes to documentation, README, etc.
- Code changes have been tested against the Discord API, or there are no code changes

### CODE REVIEW

**Comment by Jiralite** [comment]
I cannot seem to find where this is referenced:

https://github.com/discordjs/discord.js/blob/3160ab0fc75946041cbe7a093032ea10503dfd8c/packages/discord.js/src/structures/Attachment.js#L17

Perhaps it's remnants of an old refactor - possibly from being split into a builder? That property can be removed too. Pretty sure.

**Comment by Jiralite** [comment]
Okay, I just found out how and where this is being used.

Let's say a message is created and it has an attachment. They would be created as an `Attachment`. The property `this.attachment = data.url;` is being defined here.

We modify that via `AttachmentBuilder.from()`. For example: `const attachment = AttachmentBuilder.from(message.attachments.first());`

Without that property, it looks like the `AttachmentBuilder` will not be properly created. Definitely something funny is going on here, but for now, I would keep that property as it is and remove only the typing.

**Comment by net-tech** [comment]
@Jiralite Wait, revert structures change and keep the typing removed entirely or revert structures change and keep the typing but change it to `string`?

**Comment by Jiralite** [comment]
Probably best to remove the typings altogether since there is no documentation for it.

**Comment by Syjalo** [comment]
Isn't it better to make that property private?

**Comment by net-tech** [comment]
> Isn't it better to make that property private?

Done.

**Comment by Idris1401** [comment]
Maybe you should change the return type too, if it actually returns a `string` (or it doesn't matter?).

File: packages/discord.js/typings/index.d.ts [hunk 1]
```diff
@@ -1972,7 +1972,7 @@ export class AttachmentBuilder {
 
 export class Attachment {
   private constructor(data: APIAttachment);
-  public attachment: BufferResolvable | Stream;
+  private attachment: BufferResolvable | Stream;
```
Comment (kyranet) [inline packages/discord.js/typings/index.d.ts:1975]: Considering it's initially set to `data.url`, which is a string:
```suggestion
  private attachment: BufferResolvable | Stream | string;
```

**Review by kyranet — CHANGES_REQUESTED** [review]

**Review by kyranet — APPROVED** [review]

**Review by Jiralite — APPROVED** [review]
Do you mind also modifying the pull request title to reflect the changes too?

**Review by iCrawl — APPROVED** [review]

**Review by vladfrangu — APPROVED** [review]

**Review by SpaceEEC — APPROVED** [review]

**Pull request merged**